src/mpyl/steps/build/docker_build.py

```diff
@@ -70,6 +71,10 @@ def execute(self, step_input: Input) -> Output:
 
         docker_registry_config = registry_for_project(docker_config, step_input.project)
         if not step_input.dry_run:
+            if docker_registry_config.provider == "azure":
+                login(logger=self._logger, registry_config=docker_registry_config)
+            if docker_registry_config.provider == "aws":
+                login_ecr(logger=self._logger, registry_config=docker_registry_config)
             # log in to registry, because we may need to pull in a base image
             login(logger=self._logger, registry_config=docker_registry_config)
```

Comment:
you're probably still debugging this pr, but shouldn't your logic above replace this line?

Reply:
Check, thanks